### PR TITLE
Getting VSCode in sync for 1.2.0

### LINKED ISSUE #563: Extensions - for long running csharpier, use a custom tools path
Currently when the extensions keep csharpier running it prevents a user from uninstalling/updating csharpier. They get the generic error 
```bash
Failed to uninstall tool package 'csharpier': Access to the path 'C:\Users\bela\.dotnet\tools\.store\csharpier\0.13.0' is denied.
```
The VSCode extension has code in place that shuts down the csharpier process and restarts it when it is needed. But this sometimes leads to a slow format of a file. It also causes a lot of unnecessary warming of csharpier by formatting files behind the scenes.

A better approach would be to install csharpier in a custom location - https://docs.microsoft.com/en-us/dotnet/core/tools/global-tools#install-a-global-tool-in-a-custom-location

- After determining which version of csharpier should be used, check if that version of csharpier is installed in the custom location. Install it if needed.
- Always run csharpier from the custom location
- Try to determine when someone upgrades csharpier, or provide them an action to reload csharpier. This will handle the case of someone updating csharpier while the IDE is running and using the newer version.

If this approach works, then #557, #558 are not required.

### PR DESCRIPTION
**Pull request by belav**
Use DOTNET_NOLOGO #533
Use a custom tools path to allow csharpier to be updated/installed while running - fixes #563

### CODE REVIEW

File: Src/CSharpier.VSCode/package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
         "Formatters"
     ],
     "activationEvents": [
-        "onStartupFinished"
+        "onLanguage:csharp"
```
Comment (shocklateboy92) [inline Src/CSharpier.VSCode/package.json:31]: :+1:

File: Src/CSharpier.VSCode/src/CustomPathInstaller.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        if (!version) {
+            return;
+        }
+        const directoryForVersion = this.getDirectoryForVersion(version);
+        if (fs.existsSync(directoryForVersion)) {
+            this.logger.debug("File at " + directoryForVersion + " already exists");
+            return;
+        }
+
+        const command = "dotnet tool install csharpier --version " + version + " --tool-path";
```
Comment (shocklateboy92) [inline Src/CSharpier.VSCode/src/CustomPathInstaller.ts:23]: Uhm... shouldn't you have the path appended to this command after `--tool-path`?

File: Src/CSharpier.VSCode/src/CSharpierProcessProvider.ts [hunk 3]
```diff
@@ -218,9 +166,7 @@ export class CSharpierProcessProvider implements Disposable {
 
     private killRunningProcesses = () => {
         for (const key in this.csharpierProcessesByVersion) {
-            this.logger.debug(
-                "disposing of process for version " + (key === "" ? "null" : key),
-            );
+            this.logger.debug("disposing of process for version " + (key === "" ? "null" : key));
```
Comment (shocklateboy92) [inline Src/CSharpier.VSCode/src/CSharpierProcessProvider.ts:169]: nit: you can shorten this to `key || "null"`.

File: Src/CSharpier.VSCode/src/CSharpierProcessPipeMultipleFiles.ts [hunk 4]
```diff
@@ -20,9 +20,10 @@ export class CSharpierProcessPipeMultipleFiles implements ICSharpierProcess {
     }
 
     private spawnProcess = (csharpierPath: string, workingDirectory: string) => {
-        const csharpierProcess = spawn("dotnet", [csharpierPath, "--pipe-multiple-files"], {
+        const csharpierProcess = spawn(csharpierPath, ["--pipe-multiple-files"], {
```
Comment (shocklateboy92) [inline Src/CSharpier.VSCode/src/CSharpierProcessPipeMultipleFiles.ts:23]: Uh... you sure about this?
What does `csharpierPath` evaluate to? a `.exe`? a `.dll`? I don't think dotnet produces standalone executables for any platform other than windows.

File: Src/CSharpier.VSCode/src/CSharpierProcessSingleFile.ts [hunk 5]
```diff
@@ -15,9 +15,10 @@ export class CSharpierProcessSingleFile implements ICSharpierProcess {
     formatFile(content: string, filePath: string): Promise<string> {
         const directory = path.parse(filePath).dir;
         return new Promise((resolve, reject) => {
-            const csharpier = spawn("dotnet", [this.csharpierPath, "--write-stdout"], {
+            const csharpier = spawn(this.csharpierPath, ["--write-stdout"], {
```
Comment (shocklateboy92) [inline Src/CSharpier.VSCode/src/CSharpierProcessSingleFile.ts:18]: Same comment as the other place

File: Src/CSharpier.VSCode/src/CustomPathInstaller.ts (hunk 2 above)
Comment (belav) [reply Src/CSharpier.VSCode/src/CustomPathInstaller.ts:23]: er, yeah. This probably worked with my testing because I forgot to delete everything from the custom path.

File: Src/CSharpier.VSCode/src/CSharpierProcessPipeMultipleFiles.ts (hunk 4 above)
Comment (belav) [reply Src/CSharpier.VSCode/src/CSharpierProcessPipeMultipleFiles.ts:23]: It evaluates to "[path]/dotnet-csharpier", which works on windows because including the .exe is optional. I tested it out on ubuntu with when I did the rider plugin and just doublechecked it now for vscode in ubunto.

**Review by shocklateboy92 — DISMISSED** [review]
LGTM. 
But I'm not going to merge this because the version in the manifest file still ends in `-beta2`.
I'll leave you to remove it before merging. :cake:
